File: Problemset 2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int main(void)
{
    // Get text from user
    string text = get_string("Text: ");

    // Initialize required variables
    float wordCount = 1;          //To get word at the beginning of the string
    float sentenceCount = 0;
    float letterCount = 0;
    float averageSentences = 0;
    float averageLetters = 0;

    // Count Words, Sentences and Letters
    for (int i = 0; i < strlen(text); i++)
    {
        // Check for letters
        if (isupper(text[i]) || islower(text[i]))
        {
            letterCount++;
        }
        // Check for . ! ? depicting end of sentence
        else if (text[i] == 21 || text[i] == 46 || text[i] == 63 || text[i] == 33)
        {
            sentenceCount++;
        }
        // Check seperators like white space or comma or parenthesis to get words
        else if (text[i] == 32)
        {
            wordCount++;
        }
    }

    // Get Grade
    averageSentences = (sentenceCount / wordCount) * 100;
    averageLetters = (letterCount / wordCount) * 100;
    double grade = 0.0588 * averageLetters - 0.296 * averageSentences - 15.8;

    // Sort grade output
    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(grade));
    }

    // Exit Main
    return 0;
}

File: CS50x Problemset 6/sentimental-readability/readability.py
# Get User Text
userText = input("Text: ")

# Set variables
letterCount = 0
wordCount = 0
sentenceCount = 0
newWord = True

# Count letters and sentences
for char in userText:
    if ord(char) in range(65, 91) or ord(char) in range(97, 123):
        letterCount += 1
    elif ord(char) == 33 or ord(char) == 46 or ord(char) == 63:
        sentenceCount += 1

# Count words
wordCount = len(userText.split())

# Coleman-Liau index
averageLetters = (letterCount / wordCount) * 100
averageSentences = (sentenceCount / wordCount) * 100
index = (0.0588 * averageLetters) - (0.296 * averageSentences) - 15.8

# Give Result
if round(index) < 1:
    print("Before Grade 1...")
elif round(index) >= 16:
    print("Grade 16+")
else:
    print(f"Grade {round(index)}")